designs/Consolidate-NuGetConfig_OnLinux.md

```diff
@@ -0,0 +1,216 @@
+
+**3. dotnet CLI (NuGet)**
+
+  On windows, it will not be affected by this change.
+  On Linux/Mac, this change will be applied.
+
+**4. CI/CD (Headless CLI)**
+
+  On windows, it will not be affected by this change.
+  On Linux/Mac, this change will be applied. But considering most users will put the URLs in a NuGet.Config file in the repository, so the user wide NuGet.Config in both legacy path and consolidated path should be the default content in most cases. Not sure if there is any data we could use to prove this assumption.
+
+
+### Challenges
+
+**About merging NuGet.Config files in legacy path and consolidated path when they both exist**
+
+1.If the two config files are the same, NuGet will keep the one in consolidated path. 
+Shall we do the generalized comparison on the two xml files? That is, if the two functionally work the same, then they're same. So comments, space, key name differences will be ignored. It may need extra work compared with using string compare. Is this worth doing? 
+
+2.If one of them is the default NuGet.Config file, NuGet will ignore the default one and choose the other one. Will any scenarios break if we ignore the default NuGet.Config file?
+If one of them is the default NuGet.Config file(has nuget.org as the only source), the other one is not and it doesn't have nuget.org as one of it's sources.
+
+The above two should have covered most of the situations when the two config files both exist. 
+The 3rd situation is, customers use both dotnet CLI and mono, but have different contents of user wide NuGet.Config file other than default NuGet.Config file. That seems not reasonable. Is there any data we could use to prove this assumption? 
+
+**The symbolic link related .NET APIs for non-Windows platform are available in net6.0 for now, but we need to retarget some of our projects to net6.0.**
+
+There is a [Proposed API for symbolic links](https://github.com/dotnet/runtime/issues/24271)
+And the following APIs are available in net6.0:
+* [File.ResolveLinkTarget(String, Boolean) Method](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.resolvelinktarget?view=net-6.0)
+* [File.CreateSymbolicLink(String, String) Method](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.createsymboliclink?view=net-6.0)
+
+When retargeting some of our projects to net6.0, are we going to add net6.0 into the current list of TFMs, or replace some of the old TFMs?
+
+## Future Work
+Get rid of the legacy path completely. 
+
+## Open Questions
+
+**Which dotnet version are we going to bring in this change?**
+
+The version of .NET with this change will show warning to users who still have NuGet.Config in legacy path, encouraging them to move it to consolidated path.
+So it's good if more people could get this warning and get prepared to get rid of legacy path compleletly for future versions.
+For now, .NET 7 is the earlest possible version we could have this change.
```

Comment:
I don't believe this is still our plan. If we're planning to implement as soon as possible, then this "which version" question can be removed from the spec. Once the spec is merged, it will no longer provide valuable information to anyone reading it, in my opinion.  Same as the "port to net6" question below.